src/edu/ucsb/nceas/metacat/properties/AuthPropertiesDelegate.java

```diff
@@ -0,0 +1,134 @@
+/**
+ *    Purpose: A Class that handles Authentication-related properties for metacat configuration
+ */
+package edu.ucsb.nceas.metacat.properties;
+
+import edu.ucsb.nceas.utilities.FileUtil;
+import edu.ucsb.nceas.utilities.GeneralPropertyException;
+import edu.ucsb.nceas.utilities.MetaDataProperty;
+import edu.ucsb.nceas.utilities.PropertiesMetaData;
+import edu.ucsb.nceas.utilities.SortedProperties;
+
+import javax.xml.transform.TransformerException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.Vector;
+
+public class AuthPropertiesDelegate {
+    protected static final String AUTH_METADATA_FILE_NAME = "auth.properties.metadata.xml";
+    protected static final String AUTH_BACKUP_FILE_NAME = "auth.properties.backup";
+    protected static String authMetadataFilePath = null;
+    protected static PropertiesMetaData authMetaData = null;
+    protected static String siteAuthPropsFilePath = null;
+    protected static SortedProperties authBackupProperties = null;
+    private static AuthPropertiesDelegate authPropertiesDelegate;
+
+    private AuthPropertiesDelegate()
+        throws GeneralPropertyException {
+
+        authMetadataFilePath =
+            PropertyService.CONFIG_FILE_DIR + FileUtil.getFS() + AUTH_METADATA_FILE_NAME;
+        try {
+            // authMetaData holds configuration information about organization-level
+            // properties. This is primarily used to display input fields on
+            // the auth configuration page. The information is retrieved
+            // from an xml metadata file dedicated just to auth properties.
+            authMetaData = new PropertiesMetaData(authMetadataFilePath);
+
+            // The siteAuthPropsFile holds properties that were backed up
+            // the last time the auth was configured. On disk, the file
+            // will look like a smaller version of metacat.properties. It
+            // is stored in the data storage directory outside the
+            // application directories.
+            siteAuthPropsFilePath = PropertyService.getBackupDirPath().toString()
+                + FileUtil.getFS() + AUTH_BACKUP_FILE_NAME;
+            authBackupProperties = new SortedProperties(siteAuthPropsFilePath);
+            authBackupProperties.load();
+        } catch (TransformerException te) {
+            throw new GeneralPropertyException(
+                "Transform problem while loading properties: " + te.getMessage());
+        } catch (IOException ioe) {
+            throw new GeneralPropertyException("I/O problem while loading properties: " + ioe.getMessage());
+        }
+    }
+
+    protected static AuthPropertiesDelegate getInstance()
```

Comment:
Javadoc

Reply:
PropertyService, AuthPropertiesDelegate & BackupPropertiesDelegate addressed in commit 7a41ed9274d9c0e63125b03add225c38537dae91